craft_providers/lxd/lxd.py

```diff
@@ -0,0 +1,138 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""LXD command-line interface helpers."""
+
+import logging
+import pathlib
+import subprocess
+from typing import Optional
+
+from craft_providers.errors import details_from_called_process_error
+
+from .errors import LXDError
+
+logger = logging.getLogger(__name__)
+
+
+class LXD:
+    """Interface to `lxd` command-line.
+
+    :param lxd_path: Path to lxd.
+    """
+
+    def __init__(self, *, lxd_path: pathlib.Path = pathlib.Path("lxd")) -> None:
+        self.lxd_path = lxd_path
+
+    def init(self, *, auto: bool = False, sudo: bool = False) -> None:
+        """Initialize LXD.
+
+        Sudo is required if user is not in lxd group.
+
+        :param auto: Use default settings.
+        :param sudo: Use sudo to invoke init.
+        """
+        if sudo:
+            cmd = ["sudo"]
+        else:
+            cmd = []
+
+        cmd += [str(self.lxd_path), "init"]
+
+        if auto:
+            cmd.append("--auto")
+
+        try:
+            subprocess.run(cmd, check=True)
+        except subprocess.CalledProcessError as error:
+            raise LXDError(
+                "Failed to init LXD.",
+                details=details_from_called_process_error(error),
+            ) from error
+
+    def is_supported_version(self) -> bool:
+        """Check if LXD version is supported.
+
+        A helper to check if LXD meets minimum supported version for
+        craft-providers (currently >= 4.0).
+
+        :returns: True if installed version is supported.
+        """
+        version = self.version()
+        version_components = version.split(".")
+
+        try:
+            major_minor = ".".join([version_components[0], version_components[1]])
+            return float(major_minor) >= 4.0
```

Comment:
For robustness, I'd suggest comparing major and minor separately (for the case where we require the minimum version to be e.g. 4.2 and the current version is 4.10).

Reply:
Excellent point, will fix

Reply:
For now I limited it to just checking that the major version is >= 4.  If we need to bump the required version beyond 4.0, we add the minor combination check.

Reply:
`pkg_resources.parse_version` ftw
https://github.com/snapcore/snapcraft/blob/master/snapcraft/internal/elf.py#L121

Reply:
I was thinking about that earlier.  I went ahead and switched to it :D